Defer resolution of the default value of arguments used by quantize

Jack-Khuu opened the pull request:
Summary:
Quantize() (specifically GPTQ) is the sole user of the many params, but default values are introduced early and in multiple places. This is bug prone and confusing.

- For example, previously the default value of calibration tasks was [], which is not something Int8DynActInt4WeightGPTQQuantizer handles gracefully.

This diff defers default value resolution to quantize() since that is the direct call that uses them.

Differential Revision: D55458866

jerryzh168 commented on examples/models/llama2/export_llama_lib.py:
```diff
@@ -406,7 +407,11 @@ def build_args_parser() -> argparse.ArgumentParser:
+        "-G",
+        "--group_size",
+        type=int,
+        default=256,
```
since group_size doesn't make sense for other settings like fp, maybe default to None as well

Jack-Khuu replied:
We should probably do the same for the GPTQ specific args then (calibration_*)

I'll change this PR to update all of them and the lower one to just plumbing and keeping None

Jack-Khuu committed:
Defer resolution of the default value of arguments used by quantize (#2738)

Summary:

Quantize() (specifically GPTQ) is the sole user of the many params, but default values are introduced early and in multiple places. This is bug prone and confusing.
* For example, previously the default value of calibration tasks was [], which is not something `Int8DynActInt4WeightGPTQQuantizer` handles gracefully.

This diff defers default value resolution to quantize() since that is the direct call that uses them.

Reviewed By: jerryzh168

Differential Revision: D55458866

Jack-Khuu committed:
Plumb group_size to 4b quant (#2734)

Summary:

Previously group size wasn't being passed properly to 4b quant. This just passes it through

Reviewed By: mergennachin

Differential Revision: D55458352
